packages/apps/rwa-demo/src/components/AddAgentForm/AddAgentForm.tsx

```diff
@@ -37,18 +37,15 @@ export const AddAgentForm: FC<IProps> = ({ onClose }) => {
   const onSubmit = async (data: IAddAgentProps) => {
     try {
       const tx = await addAgent(data, account!);
-
       const signedTransaction = await sign(tx);
       if (!signedTransaction) return;
 
       const client = getClient();
       const res = await client.submit(signedTransaction);
 
-      console.log({ res });
-      addTransaction({
+      await addTransaction({
```

Comment:
Why do we need the await here?

Reply:
I added a function inside the addTransaction that stores the transaction to firebase (for future multisig)
and that is now async